Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -172,7 +172,16 @@ open class NavigationMapView: UIView {
         }
     }
     
-    var simulatesLocation: Bool = true
+    var simulatesLocation: Bool = true {
+        didSet {
+            if simulatesLocation {
+                mapView.location.options.puckBearingSource = .course
+                
+            } else {
+                mapView.location.options.puckBearingSource = .heading
+            }
+        }
+    }
```

Comment:
Can you add a comment explaining the logic in this property and its `didSet`? At a glance, I get the impression that the puck would be rotated based on the course during simulation (as in SimulatedLocationManager or ReplayLocationManager) but rotated based on the heading during actual navigation. Intuitively, I’d expect the puck to be rotated based on the course when it’s available, only falling back to heading otherwise.

Is the discrepancy because of tunnel simulation? I suppose in a deep tunnel, there may be no course because there’s no GPS location update, but there would likely be a heading because the magnetometer would still function normally.

(#1942 tracks rotating the camera (and puck) based on heading while walking, regardless of the location manager.)

Reply:
So right now, the `simulatesLocation` dose two thing:
1. To decide the puck bearing source. During simulated navigation, we use the course for puck bearing because we have no heading data. When the simulated navigation ends, the turn-by-turn navigation is using CLHeading( maybe we should set the puck bearing from course as default when the navigator starts? ).
2. When set up mapView, to keep the route line synced with location puck. if we're using the default `userCourseView`, the route line in navigation is listening to `mapView` location update while the `userCourseView` is using animation through route controller update. We should keep them synced by sending location update. If we're using map puck, in simulated navigation, the location update is should be triggered and sent to the map by frame.
 https://github.com/mapbox/mapbox-navigation-ios/blob/1b7ebf366baf7773d967c0805a8659b51a840c77/Sources/MapboxNavigation/NavigationMapView.swift#L340-L350

So I think the best way to implement the heading/course, and the simulated location update is to split the `simulatesLocation` into two properties. One is public to control the puck bearing source, which could adjust to tunnel check or walking situation and used by customer on their purposes. One is to control the location update from the location manager, decide whether should we manually sending location update to the location consumers.